src/routes/Search/index.js

```diff
@@ -0,0 +1,24 @@
+import { injectReducer } from '../../store/reducers'
+
+export default (store) => ({
+  path: 'search',
+  /*  Async getComponent is only invoked when route matches   */
+  getComponent (nextState, cb) {
+    /*  Webpack - use 'require.ensure' to create a split point
+     and embed an async module loader (jsonp) when bundling   */
+    require.ensure([], (require) => {
+      /*  Webpack - use require callback to define
+       dependencies for bundling   */
+      const Search = require('./containers/SearchContainer').default
+      const reducer = require('./modules/search').default
+
+      /*  Add the reducer to the store on key 'counter'  */
+      injectReducer(store, { key: 'search', reducer })
```

Comment:
why do we need injectReducer here?
It's overcomplicating the app architecture from my point of view.
Why normal React composition cannot work here?